custom_components/elmo_iess_alarm/helpers.py

```diff
@@ -67,3 +68,24 @@ async def validate_credentials(hass: core.HomeAssistant, config: dict):
     client = ElmoClient(config.get(CONF_SYSTEM_URL), domain=config.get(CONF_DOMAIN))
     await hass.async_add_executor_job(client.auth, config.get(CONF_USERNAME), config.get(CONF_PASSWORD))
     return True
+
+def generate_entity_name(entry: ConfigEntry, name: str = None) -> str:
+    pass
+    """Generate a name for the entity based on system configuration or username.
+
+    Args:
+        entry (ConfigEntry): The configuration entry from Home Assistant containing system configuration or username.
+
+    Returns:
+        str: The generated entity name, which is a combination of the domain and either the configured system name or the username.
+
+    Example:
+        >>> entry.data = {"system_name": "Seaside Home"}
+        >>> generate_entity_name(entry, "window")
+        "elmo_iess_alarm_seaside_home_window"
+    """
+    if "system_name" in entry.data:
```

Comment:
Here, instead of using directly `system_name` you can import and reuse `CONF_SYSTEM_NAME`